pkg/kubectl/cmd/config/set.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+)
+
+const (
+	cannotHaveStepsAfterError                = "Cannot have steps after %v.  %v are remaining"
+	additionStepRequiredUnlessUnsettingError = "Must have additional steps after %v unless you are unsetting it"
+)
+
+type navigationSteps []string
+
+type setOptions struct {
+	pathOptions   *pathOptions
+	unset         bool
+	propertyName  string
+	propertyValue string
+}
+
+func NewCmdConfigSet(out io.Writer, pathOptions *pathOptions) *cobra.Command {
+	options := &setOptions{pathOptions: pathOptions}
+
+	cmd := &cobra.Command{
+		Use:   "set [--unset] property-name [property-value]",
```

Comment:
`set --unset foo bar` is confusing syntax. Can we make `unset property-name` it's own command?